src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String INSERT_MANUFACTURER_REQUEST =
+            "INSERT INTO manufacturers(name, country) VALUES(?, ?)";
+    private static final String GET_MANUFACTURER_REQUEST =
+            "SELECT * FROM manufacturers WHERE is_deleted = false AND id = ?";
+    private static final String GET_ALL_MANUFACTURERS_REQUEST =
+            "SELECT * FROM manufacturers WHERE is_deleted = false";
+    private static final String UPDATE_MANUFACTURER_REQUEST =
+            "UPDATE manufacturers SET name = ?, country = ? WHERE id = ? AND is_deleted = false";
+    private static final String DELETE_REQUEST =
+            "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(INSERT_MANUFACTURER_REQUEST,
+                             PreparedStatement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create manufacturer: " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement =
+                        connection.prepareStatement(GET_MANUFACTURER_REQUEST)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (resultSet.next()) {
+                resultSet.getObject("id", Long.class);
+                String name = resultSet.getString("name");
+                String country = resultSet.getString("country");
+                Manufacturer manufacturer = new Manufacturer();
+                manufacturer.setId(id);
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
+                return Optional.of(manufacturer);
+            } else {
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer with id: " + id, e);
```

Comment:
Just for information. I would do this like :
```suggestion
                return Optional.of(manufacturer);     
        } catch (SQLException e) {
            throw new DataProcessingException("Can't get manufacturer with id: " + id, e);
         }
      return Optional.empty();
      }
```